src/Tools/dotnet-dump/Commands/SetThreadCommand.cs

```diff
@@ -1,24 +1,30 @@
 
 using Microsoft.Diagnostic.Repl;
+using System;
+using System.Collections.Generic;
 using System.CommandLine;
-using System.Threading.Tasks;
+using System.Linq;
 
 namespace Microsoft.Diagnostic.Tools.Dump
 {
-    [Command(Name = "setthread", Help = "Sets or displays the current thread id for the SOS commands.")]
+    [Command(Name = "setthread", Help = "Sets or displays the current thread for the SOS commands.")]
     [CommandAlias(Name = "threads")]
     public class SetThreadCommand : CommandBase
     {
-        [Argument(Help = "The thread id to set, otherwise displays the current id.")]
-        public int? ThreadId { get; set; } = null;
+        [Argument(Help = "The thread index to set, otherwise displays the list of threads.")]
+        public int? ThreadIndex { get; set; } = null;
 
         public AnalyzeContext AnalyzeContext { get; set; }
 
         public override void Invoke()
         {
-            if (ThreadId.HasValue)
+            if (ThreadIndex.HasValue)
```

Comment:
Then by doing this there's an interesting UX side effect: I expect as a user that if my switch fails, I still remain in the past analysis context (say if I was in thread 0, move to 30, I'd like to be left on 0 if switching to 30 fails). You never cache the past one. Also how can ThreadIndex be null, is there a way to get to invoke?